frontend/packages/client/src/hooks/useCommunityProposals.js

```diff
@@ -15,95 +11,50 @@ import {
  */
 export default function useCommunityProposals({
   communityId,
-  start = PAGINATION_INITIAL_STATE.start,
-  count = PAGINATION_INITIAL_STATE.count,
+  start: startParam = PAGINATION_INITIAL_STATE.start,
+  count: countParam = PAGINATION_INITIAL_STATE.count,
   status,
 } = {}) {
-  const [state, dispatch] = useReducer(paginationReducer, {
-    ...INITIAL_STATE,
-    pagination: {
-      ...PAGINATION_INITIAL_STATE,
-      start,
-      count,
-    },
-  });
-
-  useEffect(() => {
-    resetResults();
-  }, [status]);
+  const initialPageParam = [startParam, countParam, 0, -1];
 
   const { notifyError } = useErrorHandlerContext();
-  /**
-   * Function to fetch more results based on pagination configuration
-   */
-  const fetchMore = useCallback(async () => {
-    dispatch({ type: 'PROCESSING' });
-    const url = `${
-      process.env.REACT_APP_BACK_END_SERVER_API
-    }/communities/${communityId}/proposals?count=${
-      state.pagination.count
-    }&start=${state.pagination.start}${status ? `&status=${status}` : ''}`;
-    try {
-      const response = await fetch(url);
-      const proposals = await checkResponse(response);
 
-      const updatedProposals = proposals.data.map((p) => ({
-        ...p,
-        // missing fields added to avoid breaking the frontend rendering
-        winCount: 4480000,
-        textDecision: 'No, do not compensate them',
-      }));
+  const queryUniqueKey = ['community-proposals', String(communityId), status];
 
-      dispatch({
-        type: 'SUCCESS',
-        payload: { ...proposals, data: updatedProposals },
-      });
-    } catch (err) {
-      notifyError(err, url);
-      dispatch({ type: 'ERROR', payload: { errorData: err.message } });
-    }
-  }, [state.pagination, communityId, status, notifyError]);
+  const { isLoading, isError, data, error, fetchNextPage } = useInfiniteQuery(
+    queryUniqueKey,
+    async ({ pageParam = initialPageParam, queryKey }) => {
+      const [start, count] = pageParam;
+      const communityId = queryKey[1];
+      const url = `${
+        process.env.REACT_APP_BACK_END_SERVER_API
```

Comment:
We can use the `COMMUNITIES_URL` here and avoid using `process.env`. 

Is it possible to move this into a function in our `api/community.js` directory?

Reply:
updated